repos/system_upgrade/el7toel8/actors/kernel/checkkerneldrivers/actor.py

```diff
@@ -0,0 +1,84 @@
+    name = 'check_kernel_drivers'
+    consumes = (ActiveKernelModulesFacts, WhitelistedKernelModules, UdevAdmInfoData)
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        with open('files/removed_drivers.txt', 'r') as removed:
+            removed_drivers = []
+            whitelisted_modules = set()
+            collected_drivers = set()
+            drivers_to_report = set()
+
+            # Extracting kernel drivers from the files/removed_drivers.txt.
+            for line in removed.readlines():
+                token = line.strip()
+                if token.startswith('#') or not token:
+                    # We do not want comments or empty lines.
+                    continue
+                removed_drivers.append(token)
+
+            # Consuming whitelisted kernel modules.
+            for fact in self.consume(WhitelistedKernelModules):
+                whitelisted_modules.update(fact.whitelisted_modules)
+
+            # Collecting only non-whitelisted drivers that are part of the
+            # files/removed_drivers.txt.
+            for fact in self.consume(ActiveKernelModulesFacts):
+                for active_module in fact.kernel_modules:
+                    if active_module.filename in whitelisted_modules:
+                        continue
+                    if active_module.filename in removed_drivers:
+                        collected_drivers.add(active_module.filename)
+
+            # Going over the collected drivers and considering for reporting only
+            # those drivers that are currently used by some device.
+            udevadm_db = ''
+            for fact in self.consume(UdevAdmInfoData):
+                udevadm_db += fact.db
+            for line in udevadm_db.split('\n'):
+                if 'E: DRIVER=' in line:
+                    _, driver = line.split('=')
+                    if driver in collected_drivers:
+                        drivers_to_report.add(driver)
+
+            # In the end, we are only going to report drivers that are:
+            # - removed in the RHEL8 (are part of files/removed_drivers.txt)
+            # - not whitelisted
+            # - currently being used by some device
+            if drivers_to_report:
+                title = ('Detected loaded kernel drivers which have been removed '
+                         'in RHEL 8. Upgrade cannot proceed.')
+                URL = ('https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html-single/'
+                       'considerations_in_adopting_rhel_8/index#removed-device-drivers_hardware-enablement')
+                summary = ('Support for the following currently loaded RHEL 7 '
+                           'device drivers has been removed in RHEL 8: \n     - {}'
+                           '\nPlease see {} for details.'.format('\n     - '.join(drivers_to_report), URL))
+                remediation = ('Please disable detected kernel drivers in '
```

Comment:
@Rezney shouldn't this be now some kind of command? Or is this ok too?

Reply:
As discussed with @shaded-enmity we will be obsoleting/refactoring this. But I understand you concern. This should have probably be differentiated something like "remediation_command" vs "remediation_hint".

We have something similar even here:

https://github.com/oamg/leapp-repository/blob/38af7da7dd3355b573284ebbda825e333f74e1c4/repos/system_upgrade/el7toel8/actors/checkbrltty/actor.py#L26

So this should probably not block this PR...